docs/baremetal-operator-on-kubernetes-cluster.md

```diff
@@ -0,0 +1,179 @@
+# Running Baremetal-operator on kubernetes cluster
+
+## Description
+
+This document outlines the process of provisioning completely within
+the cluster; starting with spinning the components
+**Baremetal-operator and Ironic** over a
+**On-Premises Kubernetes Cluster**, till the **provisioning of baremetal
+machines** using only cluster resources.
+
+## Prerequisites
+
+Fully ready Kubernetes cluster (version >= 1.17)
+
+## Spinning
+
+* ### Clone the **Baremetal-operator**
+
+```console
+> git clone https://github.com/metal3-io/baremetal-operator.git
+> cd baremetal-operator
+```
+
+* ### Start **Ironic**
+
+We will be applying the ironic first as we need the cluster IP
+of ironic pod to configure it in the config-maps of Baremetal-operator for
+inter-pod communication.
+
+```console
+> kubectl apply -k ironic-deployement/default
+> kubectl get pods -n default
+```
+
+* ### Update **config maps**
+
+Take the Cluster-IP Address of ironic pod running in default namespace
+and copy it in the config maps of Baremetal-operator by replace the default
+provisioning-IP in *deploy/ironic_ci.env* as mentioned below.
```

Comment:
modifying this one is unnecessary if you deploy ironic in your cluster (as you are explaining it above). It is only used when you deplyo with run_local_ironic.sh (not in cluster)